src/Types/DateImmutableType.php

```diff
@@ -56,7 +72,27 @@ public function convertToDatabaseValue($value, AbstractPlatform $platform)
      */
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
-        if ($value === null || $value instanceof DateTimeImmutable) {
+        if ($value === null) {
+            return null;
+        }
+
+        if ($value instanceof DateTimeImmutable) {
+            $offset        = $value->format('O');
+            $defaultOffset = (new DateTimeImmutable())->format('O');
+
+            if ($offset !== $defaultOffset) {
+                Deprecation::triggerIfCalledFromOutside(
```

Comment:
Is there a realistic situation where this deprecation might be triggered? In what case could the DBAL provide timezone information to this method? If it is lost when saving to the database, this seems unlikely, but maybe there are other situations I did not think of? If the timezone of the database server and the application server differ, maybe?

Reply:
I'm not sure, I must review the flow in order to answer.